PostHog/PostHogSDK.swift

```diff
@@ -30,6 +30,7 @@ let maxRetryDelay = 30.0
     private let setupLock = NSLock()
     private let optOutLock = NSLock()
     private let groupsLock = NSLock()
+    private let featureFlagsLock = NSLock()
```

Comment:
I'd rename the lock to `flagCallReportedLock` or something like that since its not really related to the feature flags lock itself